docs/UserGuide.md

```diff
@@ -440,79 +457,113 @@ This section guides you on how to find or filter students of your choice in TAB.
 
-* You must provide at least one field to search.<br>
-  e.g. entering just `find` or `find cond/any` alone is not a valid command. You need to include the fields you wish to search for.
+* You must specify at least one field to search for, and provide at least one keyword.<br>
+  e.g. `find cond/any` or `find n/` are invalid commands.
 
-* You must provide at least one keyword to search for.<br>
-  e.g. entering just `find n/` alone is not a valid command as the keyword is empty.
+##### Search by tags
+* You can search by multiple tags. e.g. `t/paid t/new`
+* One tag parameter can only have one keyword. e.g. `t/paid new` is invalid.
+* The search is case-insensitive. e.g. keyword `new` will match `NEW`.
+* Only whole words will be matched. e.g. keyword `paid` will not match `unpaid`
 
-* Tags must only have one keyword.<br>
-  e.g. `find t/zoom math` is invalid. To search by multiple tags, you can do `find t/zoom t/math`.
+Example:
+* `find t/unpaid t/New` will return all students with both of the tags `UNPAID` and `NEW`.
 
-* The search is case-insensitive.<br>
-  e.g. keyword `hans` will match `Hans`.
+##### Search by other student fields
+* You can specify one or more keywords for each parameter. e.g. `find n/Amad Ali`.
+* A field needs to contain all specified keywords to be matched.<br>
+  e.g. keywords `Amad Ali` will not match `Amad`, but it will match `Amad bin Ali`.
+* If you specify multiple of the same prefix, only the last prefix will be used for the search of that field.
+* A keyword can match a word partially. e.g. keyword `uni` will match `university`.
+* The search is case-insensitive.
 
-* A keyword can match a word partially.<br>
-  e.g. keyword `math` will match `mathematics`.
+Example:
+* `find n/John a/west coast` will return students with name that contains `John`, and address that contains `west` and `coast`.
 
-* The order of the keywords do not matter.<br>
-  e.g. keyword `west jurong` will match `jurong west`.
-  
-* A field needs to contain all specified keywords to be matched.<br>
-  e.g. keywords `Amad Ali` will not match `Amad` or `Ali Abdul`, but it will match `Amad bin Ali`.
+##### Search by lesson fields
+* Finds all students with at least one lesson that matches the lesson parameters.
 
-Examples:
+<table id="find-param-table">
+    <thead>
+        <tr>
+            <th style="text-align:center">Fields</th>
+            <th style="text-align:center">Find behaviour</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>Subject <br> Lesson Rate</td>
+            <td>
+              <ul>
+                <li>The same constraints for student fields apply.</li>
+              </ul>
+            </td>
+        </tr>
+        <tr>
+            <td>Homework</td>
+            <td>
+              <ul>
+                <li>Finds students with lessons that have at least one homework that matches all the keywords.</li>
```

Comment:
Not sure about this phrasing/suggestion but the phrasing here seems a bit awkward

```suggestion
                <li>Finds students with lessons that have at least one piece of homework matching all the keywords.</li>
```